### ISSUE TITLE
Can HTSJDK use a VCF index to quickly count total records in a VCF?

### ISSUE THREAD

**Issue by bbimber** [samtools/htsjdk#1586]
Hello,

When working with a large VCF, iterating all features to determine the total variant count is slow. Can Can HTSJDK use a VCF index to quickly count total records in a VCF?

Thanks

**Comment by cmnbroad** [comment]
Someone else may have a more definitive answer, but I think the linear index part of a Tribble index (.idx) has that information, per-chromosome. I don't think tabix does.

**Comment by lindenb** [comment]
@cmnbroad well it should be possible as you can get this information with `bcftools index -s in.vcf.gz`

**Comment by bbimber** [comment]
exactly. i also didnt know this was possible, but bcftools apparently can do it. it would be very useful to be able to get variant count like this for big files.

**Comment by yfarjoun** [comment]
https://www.biostars.org/p/166414/#166448

bcf doesn't use a "pure" tabix index....

**Comment by lindenb** [comment]
@lindenb with a recent version of bcftools, I'm able to extract the number of variants/chrom with a **tbi** index and `bcftools index -s`.

**Comment by yfarjoun** [comment]
I'm wondering: did you make the index with tabix or bcftools?

**Comment by lindenb** [comment]
@yfarjoun bcftools. (but I think now both tools now use the same C code for tbi )

**Comment by lindenb** [comment]
@yfarjoun the C code collection metadata is here : https://github.com/samtools/htslib/blob/1d79f449cb3b02eda8fc151556395b7b50ccd730/hts.c#L2857

>  Indexes (both .tbi and .csi) made by tabix include extra data about    the indexed file.  The returns a pointer to this data.  Note that the data is stored exactly as it is in the index.  Callers need to interpret  the results themselves, including knowing what sort of data to expect  byte swapping etc.

**Comment by bbimber** [comment]
all of our indexes are made by tabix and have this info, which makes sense if bcftools/tabix share the same code
